auction/src/lib.rs

```diff
@@ -353,6 +355,10 @@ pub mod pallet {
 				!(nft.state.is_soulbound && nft.creator != nft.owner),
 				Error::<T>::CannotListNotCreatedSoulboundNFTs
 			);
+			ensure!(
+				!(nft.state.is_secret && !nft.state.is_secret_synced),
```

Comment:
Rather have a proper if statement with an enusure than this. This is hard to reason with.

Reply:
What do you mean by this ?